java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/SymbolProvider.java

```diff
@@ -0,0 +1,92 @@
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.eclipse.lsp4j.SymbolKind;
+
+import java.util.List;
+
+public interface SymbolProvider {
+    List<SymbolInformation> get(SearchMatch match) throws CoreException;
+
+    default SymbolKind convertSymbolKind(IJavaElement element) {
+        switch (element.getElementType()) {
+            case IJavaElement.TYPE:
+                try {
+                    IType type = (IType) element;
+                    if (type.isInterface()) {
+                        return SymbolKind.Interface;
+                    } else if (type.isEnum()) {
+                        return SymbolKind.Enum;
+                    }
+                } catch (JavaModelException ignore) {
+                }
+                return SymbolKind.Class;
+            case IJavaElement.ANNOTATION:
+                return SymbolKind.Property; // TODO: find a better mapping
+            case IJavaElement.CLASS_FILE:
+            case IJavaElement.COMPILATION_UNIT:
+                return SymbolKind.File;
+            case IJavaElement.FIELD:
+                IField field = (IField) element;
+                try {
+                    if (field.isEnumConstant()) {
+                        return SymbolKind.EnumMember;
+                    }
+                    int flags = field.getFlags();
+                    if (Flags.isStatic(flags) && Flags.isFinal(flags)) {
+                        return SymbolKind.Constant;
+                    }
+                } catch (JavaModelException ignore) {
+                }
+                return SymbolKind.Field;
+            case IJavaElement.IMPORT_CONTAINER:
+            case IJavaElement.IMPORT_DECLARATION:
+                //should we return SymbolKind.Namespace?
+            case IJavaElement.JAVA_MODULE:
+                return SymbolKind.Module;
+            case IJavaElement.INITIALIZER:
+                return SymbolKind.Constructor;
+            case IJavaElement.LOCAL_VARIABLE:
+                return SymbolKind.Variable;
+            case IJavaElement.TYPE_PARAMETER:
+                return SymbolKind.TypeParameter;
+            case IJavaElement.METHOD:
+                try {
+                    // TODO handle `IInitializer`. What should be the `SymbolKind`?
+                    if (element instanceof IMethod) {
+                        if (((IMethod) element).isConstructor()) {
+                            return SymbolKind.Constructor;
+                        }
+                    }
+                    return SymbolKind.Method;
+                } catch (JavaModelException e) {
+                    return SymbolKind.Method;
+                }
+            case IJavaElement.PACKAGE_DECLARATION:
+                return SymbolKind.Package;
+        }
+        return SymbolKind.String;
+    }
+
+    default Location getLocation(IJavaElement element) {
```

Comment:
I think that this has to handle the time that the location is null here, it will give everything the ability to that and it would be really nice:

https://github.com/konveyor/java-analyzer-bundle/pull/32/files#diff-56e5b92b0c6a5238cdd5025681448c0bf79a4d3ab0da5e0d43677e33cf53ec65R35-R45


something like that??

Reply:
This can be a follow-on though

Reply:
Will do :+1: